scripts/skills/effects/rf_life_leech_effect.nut

```diff
@@ -0,0 +1,100 @@
+this.rf_life_leech_effect <- ::inherit("scripts/skills/skill", {
+	m = {
+		LifeLeechValue = 1.0,   // 1.0 equals to 100% damage leeched as life
+        LeechDistance = 1,
+		ForceHeal = false   // Temporary helper variable
+	},
+	function create()
+	{
+		this.m.ID = "effects.rf_life_leech";
+		this.m.Name = "Life Leech";
+		this.m.Description = "This character recovers Hitpoints when dealing damage to enemies with blood.";
+		this.m.Icon = "skills/status_effect_09.png";
+		this.m.IconMini = "";   // We currently have no fitting mini-icon for this
+		this.m.Overlay = "status_effect_09";
+		this.m.SoundOnUse = [
+			"sounds/enemies/vampire_life_drain_01.wav",
+			"sounds/enemies/vampire_life_drain_02.wav",
+			"sounds/enemies/vampire_life_drain_03.wav"
+		];
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.Order = this.Const.SkillOrder.Perk;
+		this.m.IsActive = false;
+		this.m.IsRemovedAfterBattle = false;
+		this.m.IsStacking = false;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getTooltip();
+
+        local withinText = "adjacent enemies";
+        if (this.m.LeechDistance > 1) withinText = "enemies within " + ::MSU.Text.colorGreen(this.m.LeechDistance) + " tiles";
+        ret.push({
+            id = 11,
+            type = "text",
+            icon = "ui/icons/health.png",
+            text = "Heal " + ::MSU.Text.colorGreen((this.m.LifeLeechValue * 100) + "%") + " of Hitpoint damage inflicted on " + withinText + " that have blood"
+        });
+
+		return ret;
+	}
+
+	function onCombatStarted()
+	{
+		this.m.ForceHeal = false;
+	}
+
+	function onCombatFinished()
+	{
+		this.m.ForceHeal = false;
+	}
+
+	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		if (_damageInflictedHitpoints <= 0) return;
+		if (this.m.ForceHeal)      // If our attack killed the target we instead handle the leeching in here
+		{
+			this.lifesteal(_damageInflictedHitpoints);
+			this.m.ForceHeal = false;
+			return;
+		}
+		local actor = this.m.Container.getActor();
+		if (actor.getHitpoints() == actor.getHitpointsMax()) return;
+		if (_targetEntity == null || !_targetEntity.isAlive()) return;      // Otherwise we crash the game when leeching from a killing blow
+		if (_targetEntity.getCurrentProperties().IsImmuneToBleeding) return;
+		if (actor.getTile().getDistanceTo(_targetEntity.getTile()) > this.m.LeechDistance) return;
+
+		this.lifesteal(_damageInflictedHitpoints);
+	}
+
+	function onTargetKilled( _targetEntity, _skill )
+	{
+		if (_targetEntity.getCurrentProperties().IsImmuneToBleeding) return;
+		local actor = this.m.Container.getActor();
+		if (actor.getHitpoints() == actor.getHitpointsMax()) return;
+		if (actor.getTile().getDistanceTo(_targetEntity.getTile()) > this.m.LeechDistance) return;
+		this.m.ForceHeal = true;
+	}
+
+	function lifesteal( _damageInflictedHitpoints )
```

Comment:
Please rename the variable to something more generic e.g. `_amount`, because this function doesn't need to know if it was "damage" or anything else.

Reply:
95%+ of Lifeleech effects in other games scale from some sort of damage dealt. I think it's widely known and understood to function this way.
I will change this variable to just mean `_damageInflicted` because this may allow leeching from damage inflicted to armor or fatigue.

With something generic like `_amount` it isn't certain for me anymore what kind of value is expected to arrive into this function.

Reply:
Ok that is fine.